File: test/language/templates/cpp/StructTemplateClashOtherTypeTest.cpp
#include "gtest/gtest.h"
#include "templates/struct_template_clash_other_type/InstantiationNameClashOtherType.h"

namespace templates
{
namespace struct_template_clash_other_type
{

TEST(StructTemplateClashOtherTypeTest, readWrite)
{
    InstantiationNameClashOtherType instantiationNameClashOtherType(Test_uint32_99604043(42));

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiationNameClashOtherType.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiationNameClashOtherType readInstantiationNameClashOtherType(reader);

    ASSERT_TRUE(instantiationNameClashOtherType == readInstantiationNameClashOtherType);
}

} // namespace struct_template_clash_other_type
} // namespace templates

File: test/language/templates/python/StructTemplateClashOtherTypeTest.py
import unittest
import zserio

from testutils import getZserioApi

class StructTemplateClashOtherTypeTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "templates.zs").struct_template_clash_other_type

    def testReadWrite(self):
        instantiationNameClashOtherType = self.api.InstantiationNameClashOtherType(
            self.api.Test_uint32_99604043(42))

        writer = zserio.BitStreamWriter()
        instantiationNameClashOtherType.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiationNameClashOtherType = self.api.InstantiationNameClashOtherType()
        readInstantiationNameClashOtherType.read(reader)
        self.assertEqual(instantiationNameClashOtherType, readInstantiationNameClashOtherType)
